finality-aleph/src/sync/metrics.rs

```diff
@@ -106,26 +112,69 @@ impl Metrics {
                 )?,
             );
         }
-        Ok(Metrics::Prometheus { calls, errors })
+        Ok(Metrics::Prometheus {
+            event_calls,
+            event_errors,
+            top_finalized_block: register(
+                Counter::new("aleph_top_finalized_block", "no help")?,
+                &registry,
+            )?,
+            best_block: register(Gauge::new("aleph_best_block", "no help")?, &registry)?,
+        })
     }
 
     pub fn noop() -> Self {
         Metrics::Noop
     }
 
     pub fn report_event(&self, event: Event) {
-        if let Metrics::Prometheus { calls, .. } = self {
-            if let Some(counter) = calls.get(&event) {
+        if let Metrics::Prometheus { event_calls, .. } = self {
+            if let Some(counter) = event_calls.get(&event) {
                 counter.inc();
             }
         }
     }
 
     pub fn report_event_error(&self, event: Event) {
-        if let Metrics::Prometheus { errors, .. } = self {
-            if let Some(counter) = errors.get(&event) {
+        if let Metrics::Prometheus { event_errors, .. } = self {
+            if let Some(counter) = event_errors.get(&event) {
                 counter.inc();
             }
         }
     }
+
+    pub fn update_best_block(&self, number: BlockNumber) {
+        match self {
+            Metrics::Noop => {}
+            Metrics::Prometheus { best_block, .. } => best_block.set(number as u64),
+        }
+    }
+
+    pub fn get_best_block(&self) -> BlockNumber {
```

Comment:
this looks strange. This should be removed from the metrics code

Reply:
removed